drift-codec/src/main/java/io/airlift/drift/codec/metadata/ThriftMethodMetadata.java

```diff
@@ -242,14 +243,13 @@ public boolean isIdempotent()
 
     private ImmutableMap<Short, ThriftType> buildExceptionMap(ThriftCatalog catalog, ThriftMethod thriftMethod)
     {
-        ImmutableMap.Builder<Short, ThriftType> exceptions = ImmutableMap.builder();
+        Map<Short, ThriftType> exceptions = new HashMap<>();
         Set<Type> exceptionTypes = new HashSet<>();
 
-        if (thriftMethod.exception().length > 0) {
-            for (ThriftException thriftException : thriftMethod.exception()) {
-                exceptions.put(thriftException.id(), catalog.getThriftType(thriftException.type()));
-                checkArgument(exceptionTypes.add(thriftException.type()), "ThriftMethod [%s] exception list contains more than one value for %s", methodName(method), thriftException.type());
-            }
+        for (ThriftException thriftException : thriftMethod.exception()) {
+            ThriftType thriftType = catalog.getThriftType(thriftException.type());
+            checkArgument(exceptions.put(thriftException.id(), thriftType) == null, "ThriftMethod [%s] exception list contains multiple values for field ID [%s]", methodName(method), thriftException.id());
+            checkArgument(exceptionTypes.add(thriftException.type()), "ThriftMethod [%s] exception list contains multiple values for type [%s]", methodName(method), thriftException.type().getSimpleName());
```

Comment:
I think this might be easier to read if you have checks and then separate unconditional add/put calls.